internal/integration_test/vs/bench_allocation_tinymem.go

```diff
@@ -0,0 +1,89 @@
+package vs
+
+import (
+	"bytes"
+	_ "embed"
+	"fmt"
+	"testing"
+
+	"github.com/tetratelabs/wazero/internal/testing/require"
+)
+
+var (
+	// allocationWasm is compiled from ../../../examples/allocation/tinygo-malloc/testdata/src/greet.go
+	// We can't use go:embed as it is outside this directory. Copying it isn't ideal due to size and drift.
+	allocationTinyMemWasmPath = "../../../examples/allocation/tinygo-malloc/testdata/greet.wasm"
+	allocationTinyMemWasm     []byte
+	allocationTinyMemParam    = "wazero"
+	allocationTinyMemResult   = []byte("tinymem: wazero!")
+	allocationTinyMemConfig   *RuntimeConfig
+)
+
+func init() {
+	allocationTinyMemWasm = readRelativeFile(allocationTinyMemWasmPath)
+	allocationTinyMemConfig = &RuntimeConfig{
+		ModuleName: "greeting",
+		ModuleWasm: allocationTinyMemWasm,
+		FuncNames:  []string{"malloc_tinymem", "free_tinymem", "greeting_tinymem"},
+		NeedsWASI:  true, // Needed for TinyGo
+		LogFn: func(buf []byte) error {
+			if !bytes.Equal(allocationTinyMemResult, buf) {
+				return fmt.Errorf("expected %q, but was %q", allocationTinyMemResult, buf)
+			}
+			return nil
+		},
+	}
+}
+
+func AllocationTinyMemCall(m Module, _ int) error {
+	nameSize := uint32(len(allocationTinyMemParam))
+	// Instead of an arbitrary memory offset, use TinyGo's allocator. Notice
+	// there is nothing string-specific in this allocation function. The same
+	// function could be used to pass binary serialized data to Wasm.
+	namePtr, err := m.CallI32_I32(testCtx, "malloc_tinymem", nameSize)
+	if err != nil {
+		return err
+	}
+
+	// The pointer is a linear memory offset, which is where we write the name.
+	if err = m.WriteMemory(namePtr, []byte(allocationTinyMemParam)); err != nil {
+		return err
+	}
+
+	// Now, we can call "greeting", which reads the string we wrote to memory!
+	ptrSize, fnErr := m.CallI32I32_I64(testCtx, "greeting_tinymem", namePtr, nameSize)
+	if fnErr != nil {
+		return fnErr
+	}
+
+	// This pointer was allocated by TinyGo, but owned by Go, So, we have to
+	// deallocate it when finished
+	if err := m.CallI32_V(testCtx, "free_tinymem", namePtr); err != nil {
+		return err
+	}
+
+	// This pointer was allocated by TinyGo, but owned by Go, So, we have to
+	// deallocate it when finished
+	if err := m.CallI32_V(testCtx, "free_tinymem", uint32(ptrSize>>32)); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func RunTestAllocationTinyMem(t *testing.T, runtime func() Runtime) {
+	testCall(t, runtime, allocationTinyMemConfig, testAllocationTinyMemCall)
+}
+
+func testAllocationTinyMemCall(t *testing.T, m Module, instantiation, iteration int) {
+	err := AllocationTinyMemCall(m, iteration)
+	require.NoError(t, err, "instantiation[%d] iteration[%d] failed: %v", instantiation, iteration, err)
+}
+
+func RunTestBenchmarkAllocationTinyMem_Call_CompilerFastest(t *testing.T, vsRuntime Runtime) {
+	runTestBenchmark_Call_CompilerFastest(t, allocationTinyMemConfig, "Allocation", AllocationTinyMemCall, vsRuntime)
```

Comment:
Check the `make bench.check` target, as you may be able to get these to run automatically (e.g. in actions you can see below under 'test'. Possibly by renaming "Allocation" to something else, as I can see Allocation here, but not the new ones you made...

```
BenchmarkAllocation/Call-16
wazero-compiler 3261.21   ns/op
wasmedge        105794.42 ns/op
BenchmarkFactorial/Call-16
wazero-compiler 1117.06  ns/op
wasmedge        12121.32 ns/op
BenchmarkHostCall_CrossBoundary/Call-16
wazero-compiler 113.91  ns/op
wasmedge        2686.57 ns/op
Benchmark/memory.i32/Call-16
wazero-compiler 374.12   ns/op
wasmedge        12607.52 ns/op
Benchmark/memory.i64/Call-16
wazero-compiler 380.94   ns/op
wasmedge        13439.59 ns/op
```

Reply:
done